subprojects/docs/src/samples/signing/gnupg-signatory/build.gradle

```diff
@@ -0,0 +1,22 @@
+apply plugin: 'java'
+apply plugin: 'signing'
+
+group = 'gradle'
+version = '1.0'
+
+// Typically set in ~/.gradle/gradle.properties
+project['signing.gnupg.executable'] = null
+project['signing.gnupg.useLegacyGpg'] = "true"
+project['signing.gnupg.homeDir'] = "gnupg-home"
+project['signing.gnupg.optionsFile'] = null
+project['signing.gnupg.keyName'] = "24875D73"
+project['signing.gnupg.passphrase'] = "gradle"
+
+// START SNIPPET configure-signatory
+import org.gradle.plugins.signing.signatory.gnupg.GnupgSignatoryProvider
```

Comment:
I was wondering if we could design this differently. Instead of letting the user configure the provider we could create a new plugin e.g. `GnupgSigningPlugin` that already pre-configures everything needed. Internally, the plugin would apply the `SigningPlugin`. That would leads to less boiler plate logic, clearer boundaries between the functionality while at the same time use established semantics.